client/web/src/enterprise/batches/workspaces-list/Descriptor.tsx

```diff
@@ -29,14 +29,16 @@ export const Descriptor = <Workspace extends WorkspaceBaseFields>({
     <div className="d-flex flex-1 align-items-center pt-3 pb-3 pl-2 pr-2">
         <div className={styles.status}>{statusIndicator}</div>
         <div className="flex-1">
-            <h4 className={styles.name}>{workspace.repository.name}</h4>
-            {workspace.path !== '' && workspace.path !== '/' ? (
-                <span className={styles.path}>{workspace.path}</span>
+            <h4 className={styles.name}>{workspace?.repository.name ?? 'Workspace in hidden repository'}</h4>
+            {workspace?.path !== '' && workspace?.path !== '/' ? (
+                <span className={styles.path}>{workspace?.path}</span>
```

Comment:
Since `undefined !== '' && undefined !== '/'` is true, wouldn't this still render an empty span? I'd probably tack `workspace &&` on to the condition instead:

```suggestion
            {workspace && workspace.path !== '' && workspace.path !== '/' ? (
                <span className={styles.path}>{workspace.path}</span>
```